### PR TITLE
NEXT-33405 - Improve searchIds (performance) & createDefaultContext calls

### PR DESCRIPTION
**Pull request by ghost**
### 1. Why is this change necessary?
- AddressValidator: There is no need to use the country table and join the sales_channel & sales_channel_country tables when you can directly use the mapping table to fetch the data (this change nearly halves the DB execution time of this query)
- ThemeLifecycleService: There is no need to fetch the full MediaFolderEntity or the Parent ThemeEntity as we only need the id here
- DefaultContext: There are a few parts in the code where the context is recreated with createDefaultContext where it could be reused instead. Especially in some loops (example ProductSerializer), or the SalesChannelCreator, in which the context is unnecessarily rebuilt 6 times.

### 2. What does this change do, exactly?
- Changed the `AddressValidator` to directly fetch the `salesChannelCountryId` from the mapping table
- Changed the `AddressValidator` to use the `sales_channel_country` repository
- Changed multiple files to reduce/improve the amount of `Context::createDefaultContext()` calls
- Changed the `ThemeLifecycleService` to only fetch the required id not the entire MediaFolderEntity
- Changed the `ThemeLifecycleService` to only fetch the required id not the entire ParentThemeEntity

### 3. Describe each step to reproduce the issue or behaviour.
- Open a storefront page in the shop or execute the affected commands

### 4. Please link to the relevant issues (if any).
/

### 5. Checklist

- [X] I have rebased my changes to remove merge conflicts
- [X] I have written tests and verified that they fail without my change
- [X] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-implement-new-changelog.md) with all necessary information about my changes
- [X] I have written or adjusted the documentation according to my changes
- [X] This change has comments for package types, values, functions, and non-obvious lines of code
- [X] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: src/Core/Framework/Webhook/WebhookDispatcher.php [hunk 1]
```diff
@@ -73,13 +73,10 @@ public function dispatch(object $event, ?string $eventName = null): object
...
-        foreach ($this->eventFactory->createHookablesFor($event) as $hookable) {
-            $context = Context::createDefaultContext();
-            if ($event instanceof FlowEventAware || $event instanceof AppChangedEvent || $event instanceof EntityWrittenContainerEvent) {
-                $context = $event->getContext();
-            }
+        $context = Context::createDefaultContext();
 
-            $this->callWebhooks($hookable, $context);
+        foreach ($this->eventFactory->createHookablesFor($event) as $hookable) {
+            $this->callWebhooks($hookable, $event instanceof FlowEventAware || $event instanceof AppChangedEvent || $event instanceof EntityWrittenContainerEvent ? $event->getContext() : $context);
```
Comment (mitelg) [inline src/Core/Framework/Webhook/WebhookDispatcher.php]: this was better readable before :grin:

**Review by mitelg — COMMENTED** [review]
I really like your recent performance improvement PRs :ok_hand: very nice :rocket:

File: src/Core/Framework/Webhook/WebhookDispatcher.php (hunk 1 above)
Comment (ghost) [reply src/Core/Framework/Webhook/WebhookDispatcher.php]: I changed it again, it should be easier to read now.

**Comment by aragon999** [comment]
Very nice, I had planned to create a similar PR, regarding the multiple context creation :-) But forgot to create the PR :see_no_evil: 

I found also the [PrimaryKeyResolver.php](https://github.com/aragon999/platform/commit/4e069dfb49b7285e5bf669630b577f3766a5db2a#diff-32c92e04568b05faf71a6f5210f9abd34e9b1fab2cdf3d29f072fb7caaeac43c) and [TaskRegistry.php](https://github.com/aragon999/platform/commit/4e069dfb49b7285e5bf669630b577f3766a5db2a#diff-4d46ca9ca28ccbc2cdc2e7681462a2ccfddd064c0c82b27552df089dfb8c4756).

If you want feel free to incorporate the changes as well into your PR. Maybe there is also something more contained in there, which I missed :-)

**Comment by mitelg** [comment]
thanks for your contribution @akf-bw  👍🎉💙

**Pull request closed without merging**